include/oneapi/dpl/pstl/onedpl_config.h

```diff
@@ -49,11 +49,29 @@
 #    define _ONEDPL_PREDEFINED_POLICIES 1
 #endif
 
-#if ONEDPL_USE_TBB_BACKEND || (!defined(ONEDPL_USE_TBB_BACKEND) && !ONEDPL_USE_OPENMP_BACKEND)
+// Check availability of parallel backends
```

Comment:
Should we also check SYCL availability (`sycl\sycl.hpp` or `CL\sycl.hpp') when defining `_ONEDPL_BACKEND_SYCL` above, or maybe later?

Reply:
I added it here, because 
  - There are a lot of dependencies for that macro in this file and other files (e.g. `execution_sycl_defs.h` or `parallel_backend.h`)
  - `onedpl_config.h` is almost the first header included among other oneDPL headers (which is useful if there are many dependencies)

This is a bit inconsistent comparing to `_ONEDPL_PAR_BACKEND_TBB` or `_ONEDPL_PAR_BACKEND_OPENMP` which are now defined in `parallel_backend.h`. It is possible to fix it, but I believe it is for another PR: it requires a lot of changes.